pallets/pallet-web3-names/src/lib.rs

```diff
@@ -225,7 +218,11 @@ pub mod pallet {
 	}
 
 	#[pallet::call]
-	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+	impl<T: Config<I>, I: 'static> Pallet<T, I>
+	where
```

Comment:
I changed from `TryFrom<Vec<u8>, Error = Error<Self, I>>` to here, where the error should be representable as a pallet error, but not necessarily be a pallet error. This allow us, below, to define a new type for the dotname that does not directly depend on the runtime, hence it's much easier to unit test.